lib/vmpooler/api/v1.rb

```diff
@@ -235,6 +236,47 @@ def atomically_allocate_vms(payload)
       result
     end
 
+    def update_user_metrics(operation, vmname)
+      backend.multi
+      backend.hget("vmpooler__vm__#{vmname}", 'tag:jenkins_build_url')
+      backend.hget("vmpooler__vm__#{vmname}", 'token:user')
+      backend.hget("vmpooler__vm__#{vmname}", 'template')
+      jenkins_build_url, user, poolname = backend.exec
+
+      if user
+        user = user.gsub('.', '_')
+      else
+        user = 'unauthenticated'
+      end
+      metrics.increment("user.#{user}.#{operation}.#{poolname}")
+
+      if jenkins_build_url
+        if jenkins_build_url.include? 'litmus'
+          # Very simple filter for Litmus jobs - just count them coming through for the moment.
+          metrics.increment("usage_litmus.#{user}.#{operation}.#{poolname}")
+          return
+        end
+
+        url_parts = jenkins_build_url.split('/')[2..-1]
+        jenkins_instance = url_parts[0].gsub('.', '_')
+        value_stream_parts = url_parts[2].split('_')
+        value_stream_parts = value_stream_parts.map { |s| s.gsub('.', '_') }
+        value_stream = value_stream_parts.shift
+        branch = value_stream_parts.pop
+        project = value_stream_parts.shift
+        job_name = value_stream_parts.join('_')
+        build_metadata_parts = url_parts[3]
+        component_to_test = component_to_test('RMM_COMPONENT_TO_TEST_NAME', build_metadata_parts)
+
+        metrics.increment("usage_jenkins_instance.#{jenkins_instance}.#{value_stream}.#{operation}.#{poolname}")
+        metrics.increment("usage_branch_project.#{branch}.#{project}.#{operation}.#{poolname}")
+        metrics.increment("usage_job_component.#{job_name}.#{component_to_test}.#{operation}.#{poolname}")
+      end
+    rescue StandardError => e
+      puts 'd', "[!] [#{poolname}] failed while evaluating usage labels on '#{vmname}' with an error: #{e}"
+      raise
```

Comment:
you could consider not re-raising it here since if it fails for some reason like an unexpected `nil` value it would afect only metrics but the operation to atomically_allocate_vms would not be affected.

Reply:
Good call, no reason to raise an error and cause a VM not to checkout because of a failure to update metrics. Thanks